frontend/src/components/applyForm/widgets/MultiSelectWidget.tsx

```
@@ -0,0 +1,216 @@
+  const labelType = getLabelTypeFromOptions(options?.["widget-label"]);
+  const allOptions: ComboBoxOption[] = useMemo(() => opts ?? [], [opts]);
+
+  const [selected, setSelected] = React.useState<string[]>(toArray(value));
+
+  const comboRef = useRef<ComboBoxRef>(null);
+
+  const maxSelections = typeof maxItems === "number" ? maxItems : 3;
+  const minSelectionsEff = typeof minItems === "number" ? minItems : 0;
+  const atCap = selected.length >= maxSelections;
+
+  const isOptDisabled = useCallback(
+    (opt: ComboBoxOption) =>
+      !!enumDisabled &&
+      (enumDisabled.includes(opt.label) || enumDisabled.includes(opt.value)),
+    [enumDisabled],
+  );
+
+  const availableOptions = useMemo(
+    () =>
+      allOptions.filter(
+        (o) => !selected.includes(String(o.value)) && !isOptDisabled(o),
+      ),
+    [allOptions, selected, isOptDisabled],
+  );
+
+  const error = rawErrors.length ? true : undefined;
+  const describedBy = error
+    ? `error-for-${id}`
+    : title
+      ? `label-for-${id}`
+      : undefined;
+
+  const syncUpstream = (next: string[]) => {
+    setSelected(next);
+    onChange(
+      enumOptionsValueForIndex<S>(next, allOptions, undefined) as unknown as T,
+    );
+  };
+
+  const addByLabelOrValue = (raw: string): boolean => {
+    const txt = raw.trim();
+    if (!txt || atCap) return false;
+
+    const match =
+      allOptions.find(
+        (o) => String(o.label).toLowerCase() === txt.toLowerCase(),
+      ) ??
+      allOptions.find(
+        (o) => String(o.value).toLowerCase() === txt.toLowerCase(),
+      );
+
+    if (!match) return false;
+
+    const v = String(match.value);
+    if (selected.includes(v)) return false;
+
+    const next = [...selected, v];
+    syncUpstream(next);
+    comboRef.current?.clearSelection();
+    return true;
+  };
+
+  const removeValue = (v: string) => {
+    const next = selected.filter((s) => s !== v);
+    if (enforceMinInUI && next.length < minSelectionsEff) return;
```

are we potentially swallowing an error here? If we don't show an error and we don't update the field so that the error can get caught in validation on save, where would the user see that something is wrong?

---

good point. ^